### PR TITLE
fix(flow-helper): Cache promise rejections

### PR DESCRIPTION
**Pull request by lukehesluke**
So that failing API calls are not made repeatedly after the 1st failure

## QA

(with configs set as in https://github.com/openactive/openactive-test-suite/issues/137):

Screenshot 2020-08-17 at 16 34 34

### CODE REVIEW

File: packages/openactive-integration-tests/test/helpers/request-state.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
 const RequestHelper = require("./request-helper");
-const pMemoize = require("p-memoize");
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/helpers/request-state.js:2]: unused

File: packages/openactive-integration-tests/test/report-generator.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 const chalk = require("chalk");
 const Handlebars = require("handlebars");
-const pMemoize = require("p-memoize");
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/report-generator.js:3]: unused

File: packages/openactive-integration-tests/test/helpers/flow-helper.js [hunk 3]
```diff
@@ -20,32 +20,32 @@ class FlowHelper {
 
   getDatasetSite = pMemoize(async () => {
     return this.state.getDatasetSite();
-  });
+  }, { cachePromiseRejection: true });
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/helpers/flow-helper.js:23]: Very simple solution. Based on my understanding of the Test Suite, this shouldn't cause any issues and is in fact the behaviour that we want. Let me know what you think though as I'm still not too familiar with the project and may have overlooked something

Comment (nickevansuk) [reply packages/openactive-integration-tests/test/helpers/flow-helper.js:23]: Yep this makes sense - we only want it to be called once and cached once within the flow

**Pull request merged**